src/metabase/driver/sql/query_processor.clj

```diff
@@ -754,7 +754,21 @@
       (throw (ex-info "Summing intervals is not supported" {:args args})))
     (apply hx/call :+ (map (partial ->honeysql driver) args))))
 
-(defmethod ->honeysql [:sql :-] [driver [_ & args]] (apply hx/call :- (map (partial ->honeysql driver) args)))
+(defmethod ->honeysql [:sql :-]
+  [driver [_ & [first-arg & other-args :as args]]]
+  (assert (not (interval? first-arg))
+          "Interval as first argrument to subtraction is not allowed.")
+  (if (interval? (first other-args))
+    (do
+      (assert (every? interval? other-args)
+              "All but first argument to subtraction must contain an interval.")
+      (reduce (fn [hsql-form [_ amount unit]]
+                ;; We are adding negative amount. Inspired by `->honeysql [:sql :datetime-subtract]`.
+                (add-interval-honeysql-form driver hsql-form (- amount) unit))
+              (->honeysql driver first-arg)
+              other-args))
+    (apply hx/call :- (map (partial ->honeysql driver) args))))
+
 (defmethod ->honeysql [:sql :*] [driver [_ & args]] (apply hx/call :* (map (partial ->honeysql driver) args)))
 
 ;; for division we want to go ahead and convert any integer args to floats, because something like field / 2 will do
```

Comment:
<i class='graphite__hidden'>[Re: lines 929 to 929]<br/></i>

Here's an example of checking for valid args to functions. I'd use this style (`when` + `(throw (ex-info ...))`) for checking these preconditions. <span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/metabase/metabase/34752?utm_source=unchanged-line-comment">Graphite</a>.</span>